### ISSUE TITLE
Nested EnvironmentObject

### ISSUE THREAD

**Issue by joemasilotti** [nalexn/ViewInspector#62]
Is it not possible to test a nested `@EnvironmentObject`? The following view works when running in the simulator.

```swift
import SwiftUI

struct ParentView: View {
    @EnvironmentObject var strings: Strings
    internal let inspection = Inspection<Self>()

    var body: some View {
        NestedView()
            .environmentObject(strings)
            .onReceive(inspection.notice) { self.inspection.visit(self, $0) }
    }
}

struct NestedView: View {
    @EnvironmentObject var strings: Strings
    internal let inspection = Inspection<Self>()

    var body: some View {
        Text(strings.all[0])
            .onReceive(inspection.notice) { self.inspection.visit(self, $0) }
    }
}

class Strings: ObservableObject {
    @Published var all: [String] = []
}

let strings = Strings()
strings.all.append("New string.")
let contentView = ParentView()
    .environmentObject(strings)
```

But the following test fails.

> Thread 1: Fatal error: No ObservableObject of type Strings found. A View.environmentObject(_:) for Strings may be missing as an ancestor of this view.

```swift
import ViewInspector
@testable import App
import XCTest

class Tests: XCTestCase {
    func testNestedEnvironmentObject() throws {
        let strings = Strings()
        strings.all.append("First string")
        let view = ParentView()

        let expectation = view.inspection.inspect { view in
            let text = try view.view(NestedView.self, 0).text().string()
            XCTAssertEqual(text, "First string")
        }

        let hostedView = view.environmentObject(strings)
        ViewHosting.host(view: hostedView)
        wait(for: [expectation], timeout: 0.1)
    }
}

extension ParentView: Inspectable {}
extension NestedView: Inspectable {}
```

**Comment by nalexn** [comment]
I confirm this is currently not supported. You can test the nested view independently, but not as part of the parent hierarchy. This is a very annoying limitation, I know, looking for a way to work around it.

**Comment by joemasilotti** [comment]
Got it, thanks for the quick response!

Would you like me to leave the issue open or mark it down in the README somewhere? Maybe a new "limitations" section?

**Comment by nalexn** [comment]
Let's leave the ticket opened

**Comment by nalexn** [comment]
This just got fixed as of v0.7.0
